src/btree/bt_handle.c

```diff
@@ -181,6 +180,17 @@ __wt_btree_close(WT_SESSION_IMPL *session)
 
 	btree->bulk_load_ok = false;
 
+	/* Reset eviction information. */
+	__wt_spin_destroy(session, &btree->evict_lock);
+	btree->evict_ref = NULL;
+	btree->evict_walk_period = 0;
+	btree->evict_walk_skips = 0;
+	btree->evict_disabled = 0;
+	btree->evict_busy = 0;
+	F_CLR(btree, WT_BTREE_NO_EVICTION);
```

Comment:
This change defeats a lot of what currently makes interactions between sweep and eviction complicated, which is trying to allow ordinary eviction of trees after they are marked dead but before sweep has gotten around to discarding them.

That's okay with me: I don't know of any cases where sweep is a bottleneck, and if we find one we could explicitly wake sweep up after marking a tree dead.  I just think if we're going to make that change we should do it consciously and check the places in sweep and eviction that currently try to deal with "open-but-dead" trees.